reports/aggregators-to-improve-social-media-data-access-across-many-pods.md

```diff
@@ -0,0 +1,101 @@
+uses our [Solid Aggregator Server](https://github.com/maartyman/solid-aggregator-server) and 
+[Solid Aggregator Client](https://github.com/maartyman/solid-aggregator-client)
+to show the difference in execution time of a query
+between using the aggregator and not using the aggregator.
+We use the [SolidBench.js](https://github.com/SolidBench/SolidBench.js) benchmark to
+simulate data pods with social media data.
+
+The Solid Aggregator Server functions as an intermediate component in the Solid network. 
+The server accepts queries from client applications and 
+directly exposes the result of the queries, which are the computed bindings. 
+This allows client applications to retrieve the query results directly from the aggregator instead of 
+evaluating expensive queries themselves.
+The Solid Aggregator Client make it easier for developers to work with the aggregator.
+
+This server computes the bindings and 
+keeps them up to date when changes in the resources in the Solid network occur. 
+In other words, the server makes sure that changes in the resources are reflected in 
+the resulting bindings of a specific query. 
+We do this by re-evaluating the queries every time a resource has changed.
+
+## User flow
+
+### Actors/actresses
+
+- User of the demo
+
+### Preconditions
+
+- The user has Node.js installed.
+
+### Steps
+
+1. Clone the [demo repo](https://github.com/maartyman/solidBenchAggregatorDemo) via `git clone https://github.com/maartyman/solidBenchAggregatorDemo.git`.
+2. Install dependencies via `npm i`.
+3. Set up and start servers via `npm run setupAndStartServers`.
+4. Wait till the console says "Everything is set up and ready for the demo!".
+5. Execute one of the four demo queries in a new terminal:
+    - `npm run simple-query` (see [details](https://github.com/maartyman/solidBenchAggregatorDemo#demo-1))
+    - `npm run complex-query` (see [details](https://github.com/maartyman/solidBenchAggregatorDemo#demo-2))
+    - `npm run link-traversal-query` (see [details](https://github.com/maartyman/solidBenchAggregatorDemo#demo-3))
+    - `npm run update-example` (see [details](https://github.com/maartyman/solidBenchAggregatorDemo#demo-4))
+
+### Postconditions
+
+The user sees the results of the query in the terminal where the query was started.
+For `npm run complex-query` that is
+
+```shell
+Result aggregated (0.005 s): 
+  bindings: 
+    fr: http://localhost:3000/pods/00000000000000000296/profile/card#me
+	dist: 1
+  bindings: 
+    fr: http://localhost:3000/pods/00000000000000000318/profile/card#me
+    dist: 1
+    ...
+Result client (46.363 s): 
+  bindings: 
+    fr: http://localhost:3000/pods/00000000000000000296/profile/card#me
+	dist: 1
+  bindings: 
+	fr: http://localhost:3000/pods/00000000000000000318/profile/card#me
+	dist: 1
+	...
+```
+
+Using the aggregator it takes only 0.005s to get the results, while
+when executing the query solely in a client application it takes 46.363s.
+
+## Follow-up actions
+<!--
```

Comment:
@maartyman Are there specific follow-up actions for this challenge/demo? Maybe new challenges?

Reply:
I'm currently in the process of writing a new challenge that tackles the incremental approaches to keep the query up to date in the aggregator (https://github.com/maartyman/incremunica/tree/streaming-store).

other steps include:
- Incremental link-traversal
- Incremental schema-alignment
- Discoverability of the aggregator and combining the result of multiple aggregators to get the results to a query

Reply:
So these three steps will also be part of that challenge?

Reply:
This challenge will spawn a few other challenges. We originally only had this challenge for the whole aggregator, but we changed this because a challenge should only be around 3 months of work. At least 4 other challenges will be created:

- Instead of re-executing the query, use incremental approaches
- Incremental link-traversal
- Incremental schema-alignment
- Discoverability of the aggregator and combining the result of multiple aggregators to get the results to a query

Some of these steps will possibly be split up again in multiple challenges, as I'm not sure that for example the "Incremental schema-alignment" step can be done in 3 months.